### PR TITLE
1373 1336 support wildfly 23+ and JBoss EAP 7.4+

### PR DESCRIPTION
**Pull request by obenkenobi**
### Overview
This PR adds support for Wildfly versions 23 till latest and in turn JBoss  EAP versions 7.4 till latest. 

### Related Github Issues
- [JBoss EAP 7.4 support #1336](https://github.com/newrelic/newrelic-java-agent/issues/1336)
- [New Relic agent on Wildfly 28 #1373](https://github.com/newrelic/newrelic-java-agent/issues/1373)

### CODE REVIEW

**Comment by obenkenobi** [comment]
Re-running AITs here: https://github.com/newrelic/newrelic-java-agent/actions/runs/5955258179

**Comment by obenkenobi** [comment]
> Re-running AITs here: https://github.com/newrelic/newrelic-java-agent/actions/runs/5955258179

Since I sent a commit after I ran those AITs, the above mentioned workflow can be dismissed. AITs ran for this commit and all checks have passed.

File: instrumentation/wildfly-27/build.gradle [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    implementation('jakarta.servlet:jakarta.servlet-api:5.0.0')
+
+    implementation("io.undertow:undertow-servlet-jakarta:2.2.14.Final")
+    implementation("io.undertow:undertow-core:2.2.14.Final")
+    implementation("org.jboss.logging:jboss-logging:3.4.0.Final")
+
+}
+
+jar {
+    manifest { attributes 'Implementation-Title': 'com.newrelic.instrumentation.wildfly-27' }
```
Comment (jtduffy) [inline instrumentation/wildfly-27/build.gradle:12]: Do normally also add an alias attribute, for example: 
```
'Implementation-Title-Alias': 'wildfly-27'
```

**Review by jtduffy — APPROVED** [review]

File: instrumentation/wildfly-27/build.gradle (hunk 1 above)
Comment (obenkenobi) [reply instrumentation/wildfly-27/build.gradle]: Added `'Implementation-Title-Alias': 'wildfly'`

**Review by saulocn — APPROVED** [review]

**Pull request merged**